charmcraft/application/commands/store.py

```diff
@@ -1826,16 +1827,30 @@ def fill_parser(self, parser):
                 'The digest (remote or local) or id (local, exclude "sha256:") of the OCI image'
             ),
         )
+        parser.add_argument(
+            "--arch",
+            type=utils.ChoicesList(const.SUPPORTED_ARCHITECTURES | {"all"}),
+            help="The architectures valid for this file resource. If none are provided, the resource is uploaded without architecture information.",
+        )
 
     def run(self, parsed_args):
         """Run the command."""
         store = Store(env.get_store_config())
 
+        if parsed_args.arch:
+            if parsed_args.image:
+                raise ArgumentParsingError(
+                    "Cannot specify an architecture for an OCI image. OCI images contain architecture metadata that is used."
+                )
+            architectures = parsed_args.arch or ["all"]
```

Comment:
This parameter has to exist because of the `if parsed_args.arch:`, right?

```suggestion
            architectures = parsed_args.arch
```

Reply:
I originally had that and it broke because if `--arch` isn't called it's set to `None`. But setting a default value fixed that. Thanks for making me think about it again!

Reply:
Oh wait... I need the default to be `None` so I know it's not set  on the CLI